.github/workflows/docs-release.yml

```
@@ -73,25 +75,19 @@ jobs:
         cd docs
         make html
 
-    - name: Deploy Documentation to gh-pages
-      env:
-        GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
-      run: |
-        # Set git auth for push changes 
-        git config --global user.name "GitHub Actions"
-        git config --global user.email "41898282+github-actions[bot]@users.noreply.github.com"
-
-        # Copy the build folder to a temporary location
-        cp -R docs/build/. /tmp/build
-
-        # Checkout the gh-pages branch
-        git checkout gh-pages
+    - name: Setup Pages
+      uses: actions/configure-pages@v4
```

Why was the choice made for the version `v4` instead of `v5` ? 

In any case, please pin action by its hash, it will be like this for `v4`: 
```suggestion
      uses: actions/configure-pages@1f0c5cde4bc74cd7e1254d0cb4de8d49e9068c7d # v4.0.0
```
and same thing for actions above: 

`actions/checkout@v4` ->  `actions/checkout@11bd71901bbe5b1630ceea73d27597364c9af683 # v4.2.2`
`actions/setup-python@v5` -> `actions/setup-python@a26af69be951a213d495a4c3e4e4022e16d87065 # v5.6.0`

---

Thx, replaced, using newest commit id